### ISSUE TITLE
Protected Branches

### ISSUE THREAD

**Issue by mckelveygreg** [SouthEugeneRoboticsTeam/InfiniteRecharge-2020#10]
Make master a protected branch. 
Must be done by a repo owner or admin. 

This is to protect us from ourselves, like an accidental push to master. Happens to everyone!

https://help.github.com/en/github/administering-a-repository/configuring-protected-branches

**Comment by gnawinganimal** [comment]
Good idea 👍 
Is there any reason that we should disable merge commits?

**Comment by mckelveygreg** [comment]
Oh, like always rebase? I know some people swear by it, but I don't have an opinion there... Yet :)

I think as long as we have done protections, that will go a long way to prevent accidents

**Comment by mckelveygreg** [comment]
Looks like things are locked down! So I can't merge even because I'm not authorized, which is fine, but make sure you know who can or can't just for reference.

**Comment by andrewda** [comment]
I used to be very strictly in the "Squash and Merge" camp, but lately I've switched more to "it fully depends on the scenario". If each pull request = one feature, then "Squash and Merge" probably makes sense. However, if each pull request is a series of multiple changes (and you might want to go in and revert one of those changes later), then a regular merge or "Rebase and Merge" would be the best choice. For the latter two, you've gotta make sure the commits in each PR are nicely titled so that you don't end up with a mess of "Cleanup" commits on master.
